### PR TITLE
Implement cip-14 risk adjusted reward calculation

### PR DESCRIPTION
**Pull request by vkgnosis**
part of https://github.com/cowprotocol/services/issues/606

Implements the calculation as laid out in the CIP and sets up autopilot to use it.

### Test Plan

CI, new unit tests

### CODE REVIEW

**Review by josojo — APPROVED** [review]
Looks good.

I like how you are dealing with the many variables. Some syntax reminds me of TS :D 
( One could read all the order quotes with one DB call, but I think its not worth the additional effort)

(I have only reviewed "engineer perspective". I guess the math will be checked by Haris or Marco.)

**Review by harisang — COMMENTED** [review]
The part with the capped rewards calculations looks clean and good! Haven't checked the tests yet but will do that soon.

File: crates/autopilot/src/risk_adjusted_rewards.rs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    /// in gas units
+    pub gas_cap: f64,
+    /// in COW base units
+    pub reward_cap: f64,
+}
+
+pub struct Calculator {
+    pub config: Configuration,
+    pub database: Postgres,
+    // None means that there is no COW token on this network
```
[the remark this answers is not in the document]
Comment (vkgnosis) [reply crates/autopilot/src/risk_adjusted_rewards.rs:37]: yes

File: crates/autopilot/src/lib.rs [hunk 2]
```diff
@@ -296,6 +306,42 @@ pub async fn main(args: arguments::Arguments) {
         .native_price_estimator(&args.native_price_estimators)
         .unwrap();
 
+    let risk_adjusted_rewards = (|| {
+        if chain_id != 1 {
+            return None;
+        }
+        let cip_args = [
```
[the remark this answers is not in the document]
Comment (vkgnosis) [reply crates/autopilot/src/lib.rs:313]: I want the arguments to have different names. Otherwise it is too easy to mix them up.

**Comment by vkgnosis** [comment]
I added an e2e test using currently solvable orders. Results look reasonable to me. Will merge this to make sure it works on staging.

**Pull request merged**